lib/matplotlib/cbook.py

```diff
@@ -2550,3 +2550,61 @@ def get_label(y, default_name):
 else:
     def _putmask(a, mask, values):
         return np.copyto(a, values, where=mask)
+
+
+class Locked(object):
+    """
+    Context manager to handle locks.
+
+    Based on code from conda.
+
+    (c) 2012-2013 Continuum Analytics, Inc. / http://continuum.io
+    All Rights Reserved
+
+    conda is distributed under the terms of the BSD 3-clause license.
+    Consult LICENSE.txt or http://opensource.org/licenses/BSD-3-Clause.
+    """
+    def __init__(self, path):
+        LOCKFN = '.matplotlib_lock'
+        self.path = path
+        self.end = "-" + str(os.getpid())
+        self.lock_path = os.path.join(self.path, LOCKFN + self.end)
+        self.pattern = os.path.join(self.path, LOCKFN + '-*')
+        self.remove = True
+
+    def __enter__(self):
+        retries = 10
+        # Keep the string "LOCKERROR" in this string so that external
+        # programs can look for it.
+        lockstr = ("""\
+    LOCKERROR: It looks like conda is already doing something.
```

Comment:
Actually, I'll just take this out -- I took out the message that was displayed here anyway.  I don't think the message is terribly useful in our use case.